github/git_commits_test.go

```diff
@@ -123,6 +127,172 @@ func TestGitService_CreateSignedCommit(t *testing.T) {
 		t.Errorf("Git.CreateCommit returned %+v, want %+v", commit, want)
 	}
 }
+func TestGitService_CreateSignedCommitWithInvalidParams(t *testing.T) {
+	client, _, _, teardown := setup()
+	defer teardown()
+
+	input := &Commit{
+		SigningKey: &openpgp.Entity{},
+	}
+
+	_, _, err := client.Git.CreateCommit(context.Background(), "o", "r", input)
+	if err == nil {
+		t.Errorf("Expected error to be returned because invalid params was passed")
+	}
+}
+
+func TestGitService_CreateSignedCommitWithKey(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+	s := strings.NewReader(testKey)
+	keyring, err := openpgp.ReadArmoredKeyRing(s)
+
+	date, _ := time.Parse("Mon Jan 02 15:04:05 2006 -0700", "Thu May 04 00:03:43 2017 +0200")
+	author := CommitAuthor{
+		Name:  String("go-github"),
+		Email: String("go-github@github.com"),
+		Date:  &date,
+	}
+	input := &Commit{
+		Message:    String("m"),
+		Tree:       &Tree{SHA: String("t")},
+		Parents:    []Commit{{SHA: String("p")}},
+		SigningKey: keyring[0],
+		Author:     &author,
+	}
+
+	messageReader := strings.NewReader(`tree t
+parent p
+author go-github <go-github@github.com> 1493849023 +0200
+committer go-github <go-github@github.com> 1493849023 +0200
+
+m`)
+
+	mux.HandleFunc("/repos/o/r/git/commits", func(w http.ResponseWriter, r *http.Request) {
+		v := new(createCommit)
+		json.NewDecoder(r.Body).Decode(v)
+
+		testMethod(t, r, "POST")
+
+		want := &createCommit{
+			Message: input.Message,
+			Tree:    String("t"),
+			Parents: []string{"p"},
+			Author:  &author,
+		}
+
+		sigReader := strings.NewReader(*v.Signature)
+		signer, err := openpgp.CheckArmoredDetachedSignature(keyring, messageReader, sigReader)
+		if err != nil {
+			t.Errorf("Error verifying signature: %+v", err)
+		}
+		if signer.Identities["go-github <go-github@github.com>"].Name != "go-github <go-github@github.com>" {
+			t.Errorf("Signer is incorrect. got: %+v, want %+v", signer.Identities["go-github <go-github@github.com>"].Name, "go-github <go-github@github.com>")
+		}
+		// Nullify Signature since we checked it above
+		v.Signature = nil
+		if !reflect.DeepEqual(v, want) {
+			t.Errorf("Request body = %+v, want %+v", v, want)
+		}
+		fmt.Fprint(w, `{"sha":"s"}`)
+	})
+
+	commit, _, err := client.Git.CreateCommit(context.Background(), "o", "r", input)
+	if err != nil {
+		t.Errorf("Git.CreateCommit returned error: %v", err)
+	}
+
+	want := &Commit{SHA: String("s")}
+	if !reflect.DeepEqual(commit, want) {
+		t.Errorf("Git.CreateCommit returned %+v, want %+v", commit, want)
+	}
+}
+
+func TestGitService_createSignature_noAuthor(t *testing.T) {
+	a := &createCommit{
+		Message: String("m"),
+		Tree:    String("t"),
+		Parents: []string{"p"},
+	}
+
+	_, err := createSignature(nil, a)
+
+	if err == nil {
+		t.Errorf("Expected error to be returned because no author was passed")
+	}
+	expectedString := "Commit Author is required to sign a commit"
```

Comment:
Let's please not test against the error message value.... you can remove this section completely.
That way we can easily change it. :smile: